### PR TITLE
[WIP] RPM spec file for downstream

### PR DESCRIPTION
**Pull request by hrishin**
We are not going to merge it here. Just raised the patch here so we can review it.

### CODE REVIEW

File: openshift-pipelines-client.spec [hunk 1]
```diff
@@ -0,0 +1,81 @@
+%global debug_package %{nil}
+%global package_name openshift-pipelines-client
+%global product_name OpenShift Pipelines
+%global golang_version 1.12
+%global tkn_version 1.0.0
```
Comment (hrishin) [inline openshift-pipelines-client.spec]: Ignore the version for time being

File: openshift-pipelines-client.spec [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+Name:           %{package_name}
+Version:        %{tkn_version}
+Release:        %{tkn_release}
+Summary:        A command line tool to enhance users productivity for working with %{product_name}
+License:        ASL 2.0
+URL:            https://%{repo}/releases/tag/v%{version}
+
+Source0:        %{source_tar}
+BuildRequires:  gcc
+BuildRequires:  golang >= %{golang_version}
```
Comment (navidshaikh) [inline openshift-pipelines-client.spec]: @hrishin : I think we need following here
```
Provides:    %{package_name}
Obsoletes:   %{package_name}
```

File: openshift-pipelines-client.spec [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+Name:           %{package_name}
+Version:        %{tkn_version}
+Release:        %{tkn_release}
+Summary:        A command line tool to enhance users productivity for working with %{product_name}
+License:        ASL 2.0
+URL:            https://%{repo}/releases/tag/v%{version}
+
+Source0:        %{source_tar}
+BuildRequires:  gcc
```
Comment (sthaha) [inline openshift-pipelines-client.spec]: why is this needed?

File: openshift-pipelines-client.spec [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+%doc *.md
+%license LICENSE
+%{_bindir}/tkn
+%{_datadir}/zsh/site-functions/*
+%{_datadir}/bash-completion/completions/*
+%{_mandir}/*
+
+%package redistributable
+Summary:        %{product_name} CLI binaries for Linux, macOS and Windows
+BuildRequires:  gcc
```
Comment (sthaha) [inline openshift-pipelines-client.spec]: Does the build fail without gcc?

File: openshift-pipelines-client.spec [hunk 5]
```diff
@@ -0,0 +1,81 @@
+%global debug_package %{nil}
+%global package_name openshift-pipelines-client
+%global product_name OpenShift Pipelines
+%global golang_version 1.12
```
Comment (sthaha) [inline openshift-pipelines-client.spec]: 1.13  IIRC

File: openshift-pipelines-client.spec (hunk 2 above)
Comment (hrishin) [reply openshift-pipelines-client.spec]: @navidshaikh :+1:

File: openshift-pipelines-client.spec [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+%global repo github.com/openshift/tektoncd-cli
+%global source_dir tekton-client-%{tkn_version}-%{tkn_release}
+%global source_tar %{source_dir}.tar.gz
+
+Name:           %{package_name}
+Version:        %{tkn_version}
+Release:        %{tkn_release}
+Summary:        A command line tool to enhance users productivity for working with %{product_name}
+License:        ASL 2.0
+URL:            https://%{repo}/releases/tag/v%{version}
```
[the remark this answers is not in the document]
Comment (hrishin) [reply openshift-pipelines-client.spec]: :+1:

File: openshift-pipelines-client.spec [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+%license LICENSE
+%dir %{_datadir}/%{name}-redistributable/linux/
+%dir %{_datadir}/%{name}-redistributable/macos/
+%dir %{_datadir}/%{name}-redistributable/windows/
+%{_datadir}/%{name}-redistributable/linux/tkn-linux-amd64
+%{_datadir}/%{name}-redistributable/macos/tkn-darwin-amd64
+%{_datadir}/%{name}-redistributable/windows/tkn-windows-amd64.exe
+
+%changelog
+* Mon Sep 16 2019 Chmouel Boudjnah <chmouel@redhat.com> 0.3.1
```
Comment (sthaha) [inline openshift-pipelines-client.spec]: nit: May want to match the version above or add your entry here.

Comment (navidshaikh) [reply openshift-pipelines-client.spec]: +1, we should match the version.
Also, we can have only one entry which will be first release as RPM.

Comment (hrishin) [reply openshift-pipelines-client.spec]: :+1:
